### PR TITLE
Improves raw string handling

### PR DESCRIPTION
**Pull request by ike709**
Raw strings containing newlines no longer error. This fixes 21 errors on modern TG.

### CODE REVIEW

**Comment by ike709** [comment]
Slight issue: This doesn't handle raw strings that contain `{}` inside of them

File: OpenDreamShared/Compiler/DMPreprocessor/DMPreprocessorLexer.cs [hunk 1]
```diff
@@ -197,15 +197,56 @@ class DMPreprocessorLexer : TextLexer {
...
+                            {
+                                startIdx++;
+                                stopLength = startIdx + 1;
+                            }
+
+                            if (textBuilder[textBuilder.Length-2] == '"')
+                            {
+                                stopLength = startIdx + 2;
+                            }
+                        }
```
Comment (wixoaGit) [inline OpenDreamShared/Compiler/DMPreprocessor/DMPreprocessorLexer.cs:246]: It'd be better to instead have two separate `CreateToken` calls after building the text for each type of raw string.
`token = CreateToken(TokenType.DM_Preproc_ConstantString, text, text.Substring(2, text.Length - 3));` after line 227, and
`token = CreateToken(TokenType.DM_Preproc_ConstantString, text, text.Substring(3, text.Length - 5));` after line 219.

File: OpenDreamShared/Compiler/DMPreprocessor/DMPreprocessorLexer.cs [hunk 2]
```diff
@@ -197,15 +197,56 @@ class DMPreprocessorLexer : TextLexer {
                         textBuilder.Append('@');
                         textBuilder.Append(delimiter);
 
-                        do {
-                            c = Advance();
+                        if (delimiter == '{')
```
Comment (wixoaGit) [inline OpenDreamShared/Compiler/DMPreprocessor/DMPreprocessorLexer.cs:200]: A long raw string needs to start with `@{"`, not just `@{`

File: OpenDreamShared/Compiler/DMPreprocessor/DMPreprocessorLexer.cs [hunk 3]
```diff
@@ -197,15 +197,56 @@ class DMPreprocessorLexer : TextLexer {
...
+                        {
+                            delimiter = '}';
+                            char prev;
+                            do
+                            {
+                                prev = c;
+                                c = Advance();
+
+                                textBuilder.Append(c);
+                                if (c == '}' && prev == '\"')
```
Comment (wixoaGit) [inline OpenDreamShared/Compiler/DMPreprocessor/DMPreprocessorLexer.cs:210]: This would break with `@{"}`

**Review by wixoaGit — CHANGES_REQUESTED** [review]

**Comment by ike709** [comment]
I cba to finish this but someone should.

**Pull request closed without merging**